staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -414,7 +414,10 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 		err           error
 	)
 
-	if node.Spec.ProviderID == "" {
+	_, instancesV2Enabled := cnc.cloud.InstancesV2()
+	if node.Spec.ProviderID == "" && !instancesV2Enabled {
+		// skip the provider ID check for InstancesV2, GetInstanceProviderID only builds a ProviderID for Instances
```

Comment:
This comment should probably be above line 417

Reply:
updated!